assets/js/components/read-more/index.js

```diff
@@ -110,39 +91,55 @@ class ReadMore extends Component {
 	 * @param {obj} e event
 	 */
 	onClick( e ) {
-		const { isExpanded } = this.state;
-
 		e.preventDefault();
 
-		if ( isExpanded ) {
-			this.clampLines();
-		} else {
-			this.setState( {
-				content: this.originalContent,
-				isExpanded: ! isExpanded,
-			} );
-		}
+		const { isExpanded } = this.state;
+
+		this.setState( {
+			isExpanded: ! isExpanded,
+		} );
 	}
 
 	render() {
 		const { className } = this.props;
-		const { content } = this.state;
+		const { content, summary, clampEnabled, isExpanded } = this.state;
 
 		if ( ! content ) {
 			return null;
 		}
 
+		if ( false === clampEnabled ) {
+			return (
+				<div className={ className }>
+					<div ref={ this.reviewContent }>
+						{ content }
+					</div>
+				</div>
+			);
+		}
+
 		return (
 			<div className={ className }>
 				<div
-					ref={ this.element }
+					ref={ this.reviewSummary }
+					aria-hidden={ isExpanded }
+					style={ {
+						display: isExpanded && null !== clampEnabled ? 'none' : 'inherit',
+					} }
 					dangerouslySetInnerHTML={ {
-						// `content` is the `review` parameter returned by the `reviews` endpoint.
-						// It's filtered with `wp_filter_post_kses()`, which removes dangerous HTML tags,
-						// so using it inside `dangerouslySetInnerHTML` is safe.
-						__html: content,
+						__html: summary,
 					} }
+					data-testid="summary"
 				/>
+				<div
+					ref={ this.reviewContent }
+					aria-hidden={ ! isExpanded }
+					style={ {
+						display: ! isExpanded && null !== clampEnabled ? 'none' : 'inherit',
```

Comment:
It doesn't make any difference, but just posting it here in case you like this idea. Instead of adding a style prop, we can avoid rendering this element completely:
```diff
-<div
-	ref={ this.reviewContent }
-	aria-hidden={ ! isExpanded }
-	style={ {
-		display: ! isExpanded && null !== clampEnabled ? 'none' : 'inherit',
-	} }
->
-	{ content }
-</div>
+{ ( isExpanded || null === clampEnabled ) && (
+	<div
+		ref={ this.reviewContent }
+		aria-hidden={ ! isExpanded }
+	>
+		{ content }
+	</div>
+) }
```
Same for the div above this one.

Reply:
I had this originally, but this prevents the clamp methods seeing the clientHeight. Invisible and non-rendered devs do not have height, and we cannot read content.

Reply:
But as long as you have `null === clampEnabled` in the condition, it will be rendered. So until text is clamped, both nodes would always be visible, right?

Reply:
Possibly, will give it a try :)